github/commitstatus.go

```diff
@@ -31,6 +32,12 @@ func (e *StatusError) Error() string {
 // API is the GitHub API endpoint.
 const API = "https://api.github.com"
 
+// Maximum number of retries for the retryable http request
+const maxRetries = 3
+
+// Maximum sleep time allowed
+var maxSleepTime = 15 * time.Minute
```

Comment:
How did we come out with 15 minutes? My concern is that this will occupy a worker for 15 minutes, which can be a lot (for configurations like ours where we limit 1 task per worker). Or maybe the resource is not counted as a task ? I forgot.

My first reaction would be to use a smaller value, 5 or 10 ?

Reply:
I had a private discussion with @odormond about it.

The worker will not be occupied with cogito (resource), but it will be with pipeline setter (task).

The 15minutes because in most cases we recieved this error I noticed the rate reset time in between 10-15minutes. I was tempted to put 30min here actually. 

Remeber that build jobs can take sometimes even 1h, so if we put this value too small, I fear that this retry is usless.. 
On the other hand, we should not wait 1h to get a status check... As I wrote above, given that I usually see the rate reset time to require the waiting times up to 15minutes, I choose this value....

Personal opionion: implementing a retry for rate limit with wait time limited to 5min or 10min is way too short.